iphone/Maps/UI/PlacePage/Components/PlacePageInfoViewController.swift

```diff
@@ -114,7 +116,23 @@ class PlacePageInfoViewController: UIViewController {
 
   override func viewDidLoad() {
     super.viewDidLoad()
+    setupViews()
+  }
+
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+    guard viewDidAppear else { return }
+    // Skip data reloading on the first appearance, to avoid unnecessary updates.
+    delegate?.viewWillAppear()
```

Comment:
The fix logic looks complex, because viewWillAppear delegate is not called anymore when it appears for the first time. A reader of the code will expect that `viewWillAppear` is called during the system `viewWillAppear` event.

At least, the delegate's method should be renamed somehow (how?).

Or the old logic/name may stay, but "do not do it on the first appearance" logic should be done in PlacePageInteractor::viewWillAppear.

Or the `updateBookmarkIfNeeded` code should be made more robust.

Reply:
ViewWillApeear will be called on every appearance on the screen (viewDidLoadd is called once).

In case of the iPad it may be called several lime per the vc lifecycle.

Reply:
The logic was moved into the PlacePageInteractor to handle the first call of the `vieWillAppear`